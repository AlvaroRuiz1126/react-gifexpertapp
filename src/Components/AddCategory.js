import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({setCategories}) => {
    //el inputValue siempre sera el ultimo valor que escribio el usuario
    const [inputValue, setInputValue] = useState('');
    //para poder imprimir lo que se escribe con el teclado
    const handleValueChange = (e) => {
        //console.log(e.target.value);
        setInputValue(e.target.value);
    };

    const handleSubmit = e => {
        //evita el recargo de la pagina por defecto cuando se da enter en el formulario
        e.preventDefault();
        //console.log("Submit Hecho");
        if(inputValue.trim().length > 2){
            //el cat permite barrer o acceder a los valores dentro del state
            setCategories(cat => [inputValue, ...cat]);
            setInputValue('');
        }
    }
    
    return(
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                value={inputValue}
                onChange={handleValueChange}
            />
        </form>
    );
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategory;