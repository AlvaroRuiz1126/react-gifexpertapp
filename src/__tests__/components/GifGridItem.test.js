import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../Components/GifGridItem'

describe('Pruebas de Gif App', () => {
    const title = 'Título';
    const url = 'http://localhost.com';
    let wrapper = shallow( <GifGridItem title={title} url={url} /> );

    beforeEach(() =>{
        wrapper = shallow( <GifGridItem title={title} url={url} /> );
    });

    test('Hacer match con el component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe tener un párrafo con el título', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title); 
    });

    test('Debe ser una imagen', () => {
        const img = wrapper.find('img');
        //para ver el html del elemento encontrado
        //console.log(img.html());
        //para ver los atributos o properties
        //console.log(img.props());
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });
    
    test('Debe de tener animate_fadeIn', () => {
        const div = wrapper.find('div');
        //console.log(div.prop('className'));
        expect(true).toBe(div.prop('className').includes('animate__fadeIn'));
    });
    
});
