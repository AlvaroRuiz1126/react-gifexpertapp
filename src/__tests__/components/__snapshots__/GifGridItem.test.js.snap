// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de Gif App Hacer match con el component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Título"
    src="http://localhost.com"
  />
  <p>
    Título
  </p>
</div>
`;
